sdk/communication/communication-common/test/public/node/entraCommunicationTokenCredential.spec.ts

```diff
@@ -126,5 +261,73 @@ describe("Entra CommunicationTokenCredential", function () {
   it("Respects abort signal", () => testAbortSignal(true));
   it("Continues when abort signal isn't aborted", () => testAbortSignal(false));
 
-  // todo: doesn't retry more than 3 times
+  it("It retries only 3 times when service is busy", async function () {
+    let busy = apiMock().reply(
+      503,
+      {
+        error: "Service Unavailable",
+      },
+      {
+        "Retry-After": "0",
+      },
+    );
+
+    busy = apiMock().reply(
```

Comment:
this just overrides the mock but doesn't make it reply again. I think what you want to do is

```typescript
const busy = apiMock().times(3).reply(
      503,
      {
        error: "Service Unavailable",
      },
      {
        "Retry-After": "0",
      },
    );
```

Then the `busy.isDone()` assertion will assert that all three invocations happened.